taranis/analyze_schema.py

```diff
@@ -0,0 +1,226 @@
+)
+
+
+class AnalyzeSchema:
+    def __init__(
+        self,
+        schema_allele,
+        output,
+        remove_subset,
+        remove_duplicated,
+        remove_no_cds,
+        genus,
+        species,
+        usegenus,
+    ):
+        self.schema_allele = schema_allele
+        self.allele_name = Path(self.schema_allele).stem
+        self.output = output
+        self.remove_subset = remove_subset
+        self.remove_duplicated = remove_duplicated
+        self.remove_no_cds = remove_no_cds
+        self.genus = genus
+        self.species = species
+        self.usegenus = usegenus
+
+    def check_allele_quality(self):
+        a_quality = {}
+        allele_seq = {}
+        bad_quality_record = []
+        with open(self.schema_allele) as fh:
+            for record in SeqIO.parse(self.schema_allele, "fasta"):
+                a_quality[record.id] = {"quality": "Good quality", "reason": "-"}
+                allele_seq[record.id] = str(record.seq)
+                a_quality[record.id]["length"] = len(str(record.seq))
+                if len(record.seq) % 3 != 0:
+                    a_quality[record.id]["quality"] = "Bad quality"
+                    a_quality[record.id]["reason"] = "Can not be converted to protein"
+                    a_quality[record.id]["order"] = "-"
+                else:
+                    sequence_order = taranis.utils.check_sequence_order(str(record.seq))
+                    if sequence_order == "Error":
+                        a_quality[record.id]["quality"] = "Bad quality"
+                        a_quality[record.id]["reason"] = "Start or end codon not found"
+                        a_quality[record.id]["order"] = "-"
+                    elif sequence_order == "reverse":
+                        record_sequence = str(record.seq.reverse_complement())
+                    else:
+                        record_sequence = str(record.seq)
+                    a_quality[record.id]["order"] = sequence_order
+                    if record_sequence[0:3] not in taranis.utils.START_CODON_FORWARD:
+                        a_quality[record.id]["quality"] = "Bad quality"
+                        a_quality[record.id]["reason"] = "Start codon not found"
+                    elif record_sequence[-3:] not in taranis.utils.STOP_CODON_FORWARD:
+                        a_quality[record.id]["quality"] = "Bad quality"
+                        a_quality[record.id]["reason"] = "Stop codon not found"
+                    
+                    elif taranis.utils.find_multiple_stop_codons(record_sequence):
+                        a_quality[record.id]["quality"] = "Bad quality"
+                        a_quality[record.id]["reason"] = "Multiple stop codons found"
+                if (
+                    self.remove_no_cds
+                    and a_quality[record.id]["quality"] == "Bad quality"
+                ):
+                    bad_quality_record.append(record.id)
+
+        if self.remove_duplicated:
```

Comment:
duplicates and subset checkings must be done all times, when you use `remove_duplicated` or `remove_subset` etc.. it's only for removing them from the schema, but the statistics and que quality flag should be there indicating that there is a duplicate or a subset